features/step_wrapper_with_options.feature

```diff
@@ -0,0 +1,42 @@
+Feature: Step Wrapper with Options
+  In order to be able to write more complex step definition wrappers
+  As a developer
+  I want Cucumber to provide the "options" object to the wrapping function
+
+  Background:
+    Given a file named "features/a.feature" with:
+      """
+      Feature: Step with an option
+        Scenario: Steps
+          When I run a step with options
+      """
+    And a file named "features/step_definitions/cucumber_steps.js" with:
+      """
+      import {defineSupportCode} from 'cucumber'
+
+      defineSupportCode(({Then, When}) => {
+        When(/^I run a step with options$/, {retry: 2}, function () {
```

Comment:
I don't have a strong preference, besides the fact that {wrapperOptions: {}} introduce noice to otherwise such a nice an readable API :) .. it will also differ from the current implementation of retry in the wdio-cucumber-framework - although it doesn't have to be backward compatible.

Maybe @christian-bromann would have something more to say? 
Would you be happy with the {wrapperOptions: {retry: 2}}?